### PR TITLE
Add DefaultDict typehint for optimizer state in accelerator

### PR DESCRIPTION
**Pull request by ananthsub**
## What does this PR do?

Follow up to https://github.com/PyTorchLightning/pytorch-lightning/pull/7277
Seeing mypy failures in #7306 because of this missing annotation

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/accelerators/accelerator.py [hunk 1]
```diff
@@ -114,7 +114,7 @@ def pre_dispatch(self, trainer: 'pl.Trainer') -> None:
     def _move_optimizer_state(self) -> None:
         """ Moves the state of the optimizers to the GPU if needed. """
         for opt in self.optimizers:
-            state = defaultdict(dict)
+            state: DefaultDict = defaultdict(dict)
```
Comment (ananthsub) [inline pytorch_lightning/accelerators/accelerator.py:117]: @awaelchli @Borda @carmocca do you know why this wasn't failing in https://github.com/PyTorchLightning/pytorch-lightning/pull/7277 ?

Comment (awaelchli) [reply pytorch_lightning/accelerators/accelerator.py:117]: because mypy check is not enforced and auto merge just merged it.

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/accelerators/accelerator.py (hunk 1 above)
Comment (ananthsub) [reply pytorch_lightning/accelerators/accelerator.py:117]: Should it be enforced?

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

File: pytorch_lightning/accelerators/accelerator.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/accelerators/accelerator.py:117]: I think it should now that it's green.

cc: @Borda

**Review by carmocca — APPROVED** [review]

**Pull request merged**